migrations/245_group_role_tables__create.sql

```diff
@@ -0,0 +1,38 @@
+-- Create table and index for group roles
+CREATE TABLE IF NOT EXISTS group_roles (
+    id BIGSERIAL PRIMARY KEY,
+    role_name TEXT NOT NULL,
+    assessment_id BIGINT NOT NULL,
+    minimum INT DEFAULT 0,
+    maximum INT DEFAULT 4,
```

Comment:
I would be inclined to default `maximum` to NULL. A max of 4 seems completely arbitrary, unless there's a specific reason this is like this?

Reply:
We had the idea of having 4 as a default max since the four-role POGIL system has been widespread across a few different courses at UIUC, and we thought it would be a reasonable max for this case. However, we realize that we're trying to generalize this for any course and group role system, so having a max of NULL does make more sense. Thank you for pointing that out!

Reply:
For `sync_assessments`, we thought that having the maximum number for a role be set by default to the minimum group size would be more flexible than hard-coding the max to 4, but it seems best to be as general as possible for our implementation first before worrying. We'll make changes accordingly, thanks again!